src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/ConfigurationTests.cs

```diff
@@ -211,6 +211,61 @@ public void FullConfigurationTest(bool redact)
             Assert.Equal(CleanWhitespace(generatedConfig), CleanWhitespace(ConstructExpectedOutput(redact)));
         }
 
+        /// <summary>
+        /// This is a full configuration test that lists the configuration provider source for each piece of config.
+        /// Instead of having to explicitly define every expected value, this reuses the individual categories to ensure they
+        /// assemble properly when combined.
+        /// </summary>
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void FullConfigurationWithSourcesTest(bool redact)
+        {
+            using TemporaryDirectory contentRootDirectory = new(_outputHelper);
+            using TemporaryDirectory sharedConfigDir = new(_outputHelper);
+            using TemporaryDirectory userConfigDir = new(_outputHelper);
+
+            // Set up the initial settings used to create the host builder.
+            HostBuilderSettings settings = new()
+            {
+                Authentication = HostBuilderHelper.CreateAuthConfiguration(noAuth: false, tempApiKey: false),
+                ContentRootDirectory = contentRootDirectory.FullName,
+                SharedConfigDirectory = sharedConfigDir.FullName,
+                UserConfigDirectory = userConfigDir.FullName
+            };
+
+            settings.Urls = new[] { "https://localhost:44444" }; // This corresponds to the value in SampleConfigurations/URLs.json
+
+            // This is the settings.json file in the user profile directory.
+            File.WriteAllText(Path.Combine(userConfigDir.FullName, "settings.json"), ConstructSettingsJson(new() { "Egress.json", "CollectionRules.json"}));
+
+            // This is the appsettings.json file that is normally next to the entrypoint assembly.
+            // The location of the appsettings.json is determined by the content root in configuration.
+            File.WriteAllText(Path.Combine(contentRootDirectory.FullName, "appsettings.json"), ConstructSettingsJson(new() { "Storage.json", "Authentication.json" } ));
+
+            // This is the settings.json file in the shared configuration directory that is visible
+            // to all users on the machine e.g. /etc/dotnet-monitor on Unix systems.
+            File.WriteAllText(Path.Combine(sharedConfigDir.FullName, "settings.json"), ConstructSettingsJson(new() { "Logging.json", "Metrics.json" }));
+
+            // Create the initial host builder.
+            IHostBuilder builder = HostBuilderHelper.CreateHostBuilder(settings);
+
+            // Override the environment configurations to use predefined values so that the test host
+            // doesn't inadvertently provide unexpected values. Passing null replaces with an empty
+            // in-memory collection source.
+            builder.ReplaceAspnetEnvironment(ShowSourcesTestsConstants.DefaultProcess_EnvironmentVariables);
```

Comment:
Are we picking up any sources from environment variables? I didn't notice any in the baselines.